### ISSUE TITLE
Using cron with daemon_options causes "Can't modify frozen String error"

### ISSUE THREAD

**Issue by melissafritcher** [chef-cookbooks/chef-client#547]
### Cookbook version
9.0.4

### Chef-client version
13.6.4

### Platform Details
Ubuntu 16.04

### Scenario:
Using daemon_options: ["--run-lock-timeout] with chef-client::cron causes a  `can't modify frozen String` error

### Steps to Reproduce:
1. Clone chef-client cookbook
2. Add the following test suite to the .kitchen.yml
```
- name: cron_daemon
  run_list:
  - recipe[test::cron]
  attributes:
    chef_client:
      daemon_options: ["--run-lock-timeout 0"]
```
3. run `kitchen test cron_daemon`

### Expected Result:
Converge to succeed

### Actual Result:
```
 Compiling Cookbooks...
       
       ================================================================================
       Recipe Compile Error in /tmp/kitchen/cache/cookbooks/test/recipes/cron.rb
       ================================================================================
       
       RuntimeError
       ------------
       can't modify frozen String
       
       Cookbook Trace:
       ---------------
         /tmp/kitchen/cache/cookbooks/chef-client/recipes/cron.rb:102:in `from_file'
         /tmp/kitchen/cache/cookbooks/test/recipes/cron.rb:4:in `from_file'
       
       Relevant File Content:
       ----------------------
       /tmp/kitchen/cache/cookbooks/chef-client/recipes/cron.rb:
       
        95:    sleep_time = nil
        96:  end
        97:  env        = node['chef_client']['cron']['environment_variables']
        98:  log_file   = node['chef_client']['cron']['log_file']
        99:  append_log = node['chef_client']['cron']['append_log'] ? '>>' : '>'
       100:  
       101:  # Use daemon_options in cron.
       102>> client_bin << " #{node['chef_client']['daemon_options'].join(' ')}" if node['chef_client']['daemon_options'].any?
       103:  
       104:  # If "use_cron_d" is set to true, delete the cron entry that uses the cron
       105:  # resource built in to Chef and instead use the cron_d LWRP.
       106:  if node['chef_client']['cron']['use_cron_d']
       107:    cron 'chef-client' do
       108:      action :delete
       109:    end
       110:  
       111:    cron_d 'chef-client' do
       
```

**Comment by melissafritcher** [comment]
I am working on a PR. Will submit in an hour or so.

**Comment by tas50** [comment]
Thanks for working on a PR

**Comment by tas50** [comment]
Fixed in 10.0
